pandas/core/arrays/datetimes.py

```diff
@@ -2035,11 +2034,7 @@ def _sequence_to_dt64ns(
     if out_unit is not None:
         out_dtype = np.dtype(f"M8[{out_unit}]")
 
-    if (
-        data_dtype == object
-        or is_string_dtype(data_dtype)
-        or isinstance(data_dtype, SparseDtype)
```

Comment:
Sparse is no longer necessary?

Reply:
it gets ruled out by the ensure_arraylike_for_datetimelike call on L2023